doc/cabal-package.rst

```diff
@@ -801,18 +800,45 @@ Library
 
     Build information for libraries.
 
-    A package can have one default library, which is identified by omitting the
-    ``name`` argument.
+    A package can include zero or more library components, being an optional
+    'main' unnamed library (which is identified by having no ``name`` argument)
+    and/or zero or more 'subsidiary' or 'ancilliary' named libraries (referred
+    to as 'sublibraries', which are identified by their ``name`` argument). A
+    sublibrary cannot have the same name as its package.
 
-    Starting with Cabal 2.0, sub-library components can be defined by setting
-    the ``name`` field to a name different from the current package's name; see
-    section on :ref:`Sublibraries <sublibs>` for more information. By
-    default, these sub-libraries are private and internal. Since Cabal 3.0,
-    these sub-libraries can also be exposed and used by other packages. See the
-    :pkg-field:`library:visibility` field and :ref:`Multiple Public Libraries
-    <publicsublibs>` for more information.
+    A sublibrary has a 'visibility' which can be 'public' or 'private' (the
+    default) - see the :pkg-field:`library:visibility` field. The 'main'
+    unnamed library is always public. A private sublibrary may be referred to as
+    an 'internal' library, because other packages cannot depend upon it.
```

Comment:
```suggestion
    A sublibrary may be marked 'public' or 'private' (the
    default) — see the :pkg-field:`library:visibility` field. The main
    library is always public. A private sublibrary is also known as
    an internal library. 
    Only public library components can be depended on by other packages.
```
I'm suggesting:
  - only quoting things when necessary
  - changing a dash to an em-dash
  - dropping the reference to "unnamed". I don't think that's relevant here
  - split the last sentence into two.

Reply:
There has been debate about whether library components should be referred to as (a) unnamed/named or (b) main/sub[sidiary]. The former is objective, but does not convey the relative 'standing' of the named library component. The latter conveys the (subjective) relative 'standing'. I avoided that choice by using "'main' unnamed library".

The quotation marks around the other words on their first introduction were also because they were being used figuratively rather than literally. An alternative would be to use code font (eg ``visibility``).